### PR TITLE
Merge Staging to Master

### PR DESCRIPTION
**Pull request by pa1nki113r**

### CODE REVIEW

**Review by iamcarrotmaster — APPROVED** [review]

**Review by JMartinez9820 — APPROVED** [review]
I've downloaded it along with the new changes I threw in, and it seems to be safe to merge for now. Going to wait on other's approval to see if works.

**Pull request merged**
